src/Deinterleave.cpp

```diff
@@ -578,173 +578,247 @@ class Interleaver : public IRMutator {
     }
 
     HALIDE_NEVER_INLINE Stmt gather_stores(const Block *op) {
-        const LetStmt *let = op->first.as<LetStmt>();
-        const Store *store = op->first.as<Store>();
-
-        // Gather all the let stmts surrounding the first.
-        std::vector<Stmt> let_stmts;
-        while (let) {
-            let_stmts.emplace_back(let);
-            store = let->body.as<Store>();
-            let = let->body.as<LetStmt>();
+        // First do a pass over all child stmts of this block nest
+        // looking for candidate buffers and strides.
+        struct Candidate {
+            std::string buffer;
+            int stride;
+            Type type;
+            Expr base;
+        };
+        std::vector<Candidate> candidates;
+        std::vector<Stmt> pending;
+        pending.emplace_back(op);
+        while (!pending.empty()) {
+            Stmt next = std::move(pending.back());
+            pending.pop_back();
+            if (const Block *b = next.as<Block>()) {
```

Comment:
Similarly to the previous comment about block_to_vector, it seems that block ordering may be lost here.